pkg/api/v1/defaults.go

```diff
@@ -389,3 +389,31 @@ func SetDefaults_HostPathVolumeSource(obj *v1.HostPathVolumeSource) {
 		obj.Type = &typeVol
 	}
 }
+
+func SetDefaults_SecurityContext(sc *v1.SecurityContext) {
+	ape := true
+	if sc == nil {
```

Comment:
how does this interact with Convert_v1_PodSpec_To_api_PodSpec and SetDefaults_PodSpec?

Reply:
that is a good question that I do not really know the answer too, I am not really well versed in the magic behind how api machinery works or the order of operations on which defaults are called

Reply:
cc @pmorie for feedback on pod/container securitycontext interactions for defaulting, overriding, and effective permissions...